include/deal.II/fe/fe_values.h

```diff
@@ -1663,9 +1687,9 @@ class FEValuesBase : protected FEValuesData<dim,spacedim>,
    *
    * @dealiiRequiresUpdateFlags{update_values}
    */
-  template <class InputVector, typename number>
+  template <class InputVector>
   void get_function_values (const InputVector &fe_function,
-                            std::vector<number> &values) const;
+                            std::vector<typename InputVector::value_type> &values) const;
```

Comment:
i gave it a quick try, there are numerous follow-up errors like

```
error: assigning to 'value_type' (aka 'double') from incompatible type 'value_type' (aka 'dealii::ProductType<double, double>')
```
and
```
/Users/davydden/libs-sources/deal.ii/davydden/include/deal.II/base/tensor_base.h:1006:17: error: cannot convert 'const dealii::ProductType<double, double>' to 'double' without a conversion operator
    values[i] = Number(p.values[i]);
```

Unless it is obvious how to fix it (supposedly something is missing in `ProductType` class), i would leave it for another PR and add `todo` in documentation of the function, so that it's not forgotten. 

Otherwise adjusting the code to use ProductType should rather be mechanical.